engine/src/main.rs

```diff
@@ -220,5 +223,39 @@ async fn run_main(settings: Settings) -> anyhow::Result<()> {
 		}
 		.boxed()
 	})
-	.await
+	.await;
+
+	if let Err(e) = root_result {
+		let sc_error = e.downcast_ref::<CreateBlockStreamError>();
+		match sc_error {
+			Some(CreateBlockStreamError::NoLongerCompatible {
+				cfe_version,
+				cfe_version_required,
+				first_incompatible_block,
+			}) => {
+				tracing::info!(
+					"Chainflip Engine is no longer compatible with the current state chain. \
+					Engine version: {}, required version: {}, first incompatible block: {}",
+					cfe_version,
+					cfe_version_required,
+					first_incompatible_block
```

Comment:
FWIW I think thiserror defines display methods so I don't think you need to deconstruct the errors here.

Reply:
It's setting up for a future PR where this will convert the error to something readable by the runner